src/js/site/nav.js

```diff
@@ -52,22 +52,53 @@ $(document).ready(function () {
   });
 })();
 
-// Open dropdown in language selector
+/**
+ * openLangDropdown handles setting up the toggling of our language picker
+ * on mobile and on desktop.
+ * */
 const openLangDropdown = () => {
-  const globeBtn = $('#globe-lang-btn');
-  const langPicker = $('#lang-picker');
+  const globeBtn = document.getElementById('globe-lang-btn');
+  const langPicker = document.getElementById('lang-picker');
+  const underlyingMobileMenu = document.querySelector('.mobile-sidebar');
 
-  $('body').on('mouseup', (e) => {
-    globeBtn.on('click', () => {
-      langPicker.toggleClass('lang-active');
-    });
+  // These vars are specific to the mobile picker.
+  const dimmer = document.querySelector('.lang-picker-mobile-dimmer');
+  const mobileBackBtn = document.getElementById('lang-picker-back-button');
+  const mobileCloseBtn = document.getElementById('lang-picker-close-button');
+  const globalNav = document.querySelector('.global-nav--nav');
+
+  // open the menu
+  globeBtn.addEventListener('click', () => {
+    langPicker.classList.toggle('lang-active');
+    underlyingMobileMenu.classList.add('hidden-md');
+    dimmer.style.display = 'inherit';
+    globalNav.classList.add('lang-picker-open');
+  });
+
+  // enable going 'back' to the mobile menu.
+  mobileBackBtn.addEventListener('click', () => {
+    langPicker.classList.remove('lang-active');
+    underlyingMobileMenu.classList.remove('hidden-md');
+    dimmer.style.display = 'none';
+    globalNav.classList.remove('lang-picker-open');
+  });
+
+  // closes the lang picker, restoring the mobile menu.
+  mobileCloseBtn.addEventListener('click', () => {
+    underlyingMobileMenu;
+    langPicker.classList.remove('lang-active');
+    underlyingMobileMenu.classList.remove('hidden-md');
+    dimmer.style.display = 'none';
+    // simulate a click to close the menu.
+    document.querySelector('.global-nav--toggle').click();
+    globalNav.classList.remove('lang-picker-open');
+  });
 
-    if (
-      !langPicker.is(e.target) &&
-      langPicker.has(e.target).length === 0 &&
-      langPicker.hasClass('lang-active')
-    ) {
-      langPicker.removeClass('lang-active');
+  $('body').on('click', (e) => {
+    if (!globeBtn.contains(e.target) && !langPicker.contains(e.target)) {
+      underlyingMobileMenu.classList.remove('hidden-md');
+      langPicker.classList.remove('lang-active');
+      dimmer.style.display = 'none';
     }
```

Comment:
Why are we attaching a click event on `body`?

Reply:
There is a bug in prod when clicking on the lang picker glob icon. To recreate:

1. Load the page.
2. click anywhere *but* the globe
3. click on the globe to open the language picker

(It won't open).
The code causing this is [here](https://github.com/circleci/circleci-docs/blob/master/src/js/site/nav.js#L60-L73).

Reply:
@teesloane Thanks for the added context. I'm struggling to understand why this code even exists in the first place?